pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -319,7 +320,8 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
                 'epoch':                     training epoch
                 'global_step':               training global step
                 'pytorch-lightning_version': The version of PyTorch Lightning that produced this checkpoint
-                'callbacks':                 "callback specific state"[] # if not weights_only
+                'callbacks':                                  # if not weights_only
+                    {callback.state_key: callback.on_save_checkpoint or callback.state_dict}
```

Comment:
```suggestion
```
this comment does not have enough context for someone new reading this to understand what this means. i would recommend no adding this here, but rather elsewhere in the checkpoint connector closer to where this switch happens